sky/spot/recovery_strategy.py

```diff
@@ -119,35 +186,13 @@ def launch(self,
                 'seconds.')
             time.sleep(gap_seconds)
 
-    def recover(self) -> float:
-        """Relaunch the spot cluster after failure and wait until job starts.
-
-        When recover() is called the cluster should be in STOPPED status (i.e.
-        partially down).
-
-        Returns: The timestamp job started.
-        """
-        raise NotImplementedError
-
-    def terminate_cluster(self):
-        """Terminate the spot cluster."""
-        handle = global_user_state.get_handle_from_cluster_name(
-            self.cluster_name)
-        if handle is not None:
-            try:
-                self.backend.teardown(handle, terminate=True)
-            except RuntimeError:
-                logger.error(
-                    f'Failed to terminate the spot cluster {self.cluster_name}.'
-                )
-
 
 class FailoverStrategyExecutor(StrategyExecutor, name='FAILOVER', default=True):
     """Failover strategy: wait in same region and failover after timout."""
 
     _MAX_RETRY_CNT = 240  # Retry for 4 hours.
-    _RETRY_INIT_GAP_SECONDS = 60
 
+    @_maybe_retry_until_up
```

Comment:
I'm confused why wrapping this whole thing with retry, because Steps 1 to 2 include stuff that shouldn't need to be retried?  Is it possible to only wrap the provisioning steps with retry?